MapboxNavigation/RouteMapViewController.swift

```diff
@@ -395,124 +320,148 @@ class RouteMapViewController: UIViewController {
+                let lookAheadDistance: CLLocationDistance = 10
+                guard let pointAheadFeature = featurePolyline.sliced(from: closestCoordinate).coordinateFromStart(distance: lookAheadDistance) else { continue }
+                guard let pointAheadUser = slicedLine.coordinateFromStart(distance: lookAheadDistance) else { continue }
+                guard let reversedPoint = Polyline(featureCoordinates.reversed()).sliced(from: closestCoordinate).coordinateFromStart(distance: lookAheadDistance) else { continue }
 
-        self.mapView.enableFrameByFrameCourseViewTracking(for: duration)
-        self.mapView.setNeedsUpdateConstraints()
+                let distanceBetweenPointsAhead = pointAheadFeature.distance(to: pointAheadUser)
+                let distanceBetweenReversedPoint = reversedPoint.distance(to: pointAheadUser)
+                let minDistanceBetweenPoints = min(distanceBetweenPointsAhead, distanceBetweenReversedPoint)
 
-        let animate = {
-            self.view.layoutIfNeeded()
-            self.navigationView.floatingStackView.alpha = 1.0
-        }
+                if minDistanceBetweenPoints < smallestLabelDistance {
+                    smallestLabelDistance = minDistanceBetweenPoints
 
-        let complete: (Bool) -> Void = {
-            self.navigationView.endOfRouteView?.isHidden = true
-            self.unembedEndOfRoute()
-            completion?($0)
+                    if let line = feature as? MLNPolylineFeature {
+                        let roadNameRecord = self.roadFeature(for: line)
+                        currentShieldName = roadNameRecord.shieldName
+                        currentName = roadNameRecord.roadName
+                    } else if let line = feature as? MLNMultiPolylineFeature {
+                        let roadNameRecord = self.roadFeature(for: line)
+                        currentShieldName = roadNameRecord.shieldName
+                        currentName = roadNameRecord.roadName
+                    }
+                }
+            }
         }
 
-        let noAnimation = {
-            animate()
-            complete(true)
+        let hasWayName = currentName != nil || currentShieldName != nil
+        if smallestLabelDistance < 5, hasWayName {
+            if let currentShieldName {
+                self.navigationView.wayNameView.attributedText = currentShieldName
+            } else if let currentName {
+                self.navigationView.wayNameView.text = currentName
+            }
+            self.navigationView.wayNameView.isHidden = false
+        } else {
+            self.navigationView.wayNameView.isHidden = true
         }
-
-        guard duration > 0.0 else { return noAnimation() }
-        UIView.animate(withDuration: duration, delay: 0.0, options: [.curveLinear], animations: animate, completion: complete)
     }
+	
+    // MARK: End Of Route
+	
+    func transitionToEndNavigation(with duration: TimeInterval, completion: ((Bool) -> Void)? = nil) {
```

Comment:
Did anything change here or was this simply moved?

Reply:
this is new